### ISSUE TITLE
Cyberpunk 2077 Xid 31

### ISSUE THREAD

**Issue by valgusk** [HansKristian-Work/vkd3d-proton#1510]
Before and after #1507 fix was available, game reliably freezes with Xid 31 ACCESS_TYPE_VIRT_READ. 
Having RT/Esync/Fsync Off can delay the issue, but ultimately the freeze is guaranteed.
Before considerably increasing `vm.max_map_count` it happened much faster, at times instantly after start of gameplay. Now it can survive up to a couple of minutes of gameplay.
Also tried older driver version (520.xx), but it only made matters worse.

### Software information
Cyberpunk 2077, DLSS Enabled, PT Enabled

### System information
- GPU: RTX 2060S
- Driver: 530.41.03
- Wine version: Proton: 1681390027 experimental-bleeding-edge-7.0-40513-20230413-pac650e-w1dd2c7-d01dc9d-vccb744
- VKD3D-Proton version: 

### Log files
[journalctl](https://gist.github.com/valgusk/56034c84b3450a3dee982a4b21a436d3)
[steam-1091500.log](https://gist.github.com/valgusk/26642b696a505622459caca64f22652b)
[vkd3d.log](https://gist.github.com/valgusk/3f31563374ed43e9f3bfe4da7f6d16a4)
[ulimits](https://gist.github.com/valgusk/63507d2f2283ea66c338aca7ce3a8183)
[sysctl -a](https://gist.github.com/valgusk/bb29b7b944b15581089828515b0cc08f)
[Steam system info](https://gist.github.com/valgusk/1e83debeda14f4c171acb242c849be6f)

**Comment by noobynoobs73** [comment]
Got this issue as well, with a RTX 2070. Did you find a workaround soluiton ? Freese appears within seconds of starting playing.

**Comment by valgusk** [comment]
> Got this issue as well, with a RTX 2070. Did you find a workaround soluiton ? Freese appears within seconds of starting playing.

@noobynoobs73  
Sorry, but no, as of now the only workaround I know is to use Windows. I finished the game already on the older version, when it worked, but if you didn't, keep in mind that Windows also has at least 2x better performance. In VKD3D-Proton I get sub-10 fps, while on native D3D I get 20-30fps with PT 1080p and DLSS, so mostly playable.

**Comment by VeilSilence** [comment]
535 beta nvidia fixed some Xid errors.

**Comment by runar-work** [comment]
When I first looked into it, the freezes happened randomly after a few minutes of playing, as described. Unfortunately, the breadcrumb traces were completely random too, with different shaders and such indicated every time, so no luck there.

Then I did some more testing last week and got consistent freezes when loading autosaves in the nomad mission, after heading out in the car and before reaching Night City. It happens when the loading progress bar reaches around 60%. Same Xid too. The four breadcrumb traces I got are more consistent here, so hopefully a bit more useful than the first case.

- GPU: GTX 1660 Ti
- Driver: 535.54.3
- Wine version: Proton Experimental bleeding-edge

```NVRM: Xid (PCI:0000:08:00): 31, pid=11922, name=GameThread, Ch 000000ee, intr 00000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_GCC faulted @ 0x38b0_00000000. Fault is of type FAULT_PDE ACCESS_TYPE_VIRT_READ```

[cyberpunk_Xid31_logs.zip](https://github.com/HansKristian-Work/vkd3d-proton/files/11881689/cyberpunk_Xid31_logs.zip)
[cyberpunk_autosave_border_crossing.zip](https://github.com/HansKristian-Work/vkd3d-proton/files/11881693/cyberpunk_autosave_border_crossing.zip)

**Comment by runar-work** [comment]
Hm, doesn't seem to be entirely the same issue. I tested with 530 and the autosave loads fine, so this is an added regression.

**Comment by xpander69** [comment]
with 525.47.27 (vulkan beta) and proton-experimental (and BE) i had both xid31 and xid109 within 10-20 min of playing.
switched to 535.54.03 and played 4 hours in a row without any issues. autosaves and stuff load also fine, but i already finished the game starting as nomad with the previous patch (1.61) playthrough, so i cant confirm if that side has some issues. Currently on Corpo run.
GPU: RTX 3080

**Comment by chrishewlings** [comment]
Still throws Xid 31 and freezes within ~30 seconds of starting gameplay here with Fedora 37, RTX 2070S, driver version 535.54.03, Proton Experimental.

**Comment by chrishewlings** [comment]
Update: switched back from Proton Experimental -> Proton GE 7.55 and now it seems to have resolved the crash.

**Comment by Dahipppo1555** [comment]
Well nothing helped.
CP77 ver 1.63
VKD3D 2.9
Nvidia 535-86-05

and still getting Xid 31

**Comment by HamilcarR** [comment]
> Update: switched back from Proton Experimental -> Proton GE 7.55 and now it seems to have resolved the crash.

I did the same and that's what I thought , but if you play longer , there will definitely be a crash. 
Yesterday I had a very long session of 2 hours , but it still ended up with a crash.

**Comment by StevenPartridge** [comment]
Same issue on Garuda Lixux, 2080 Super, 535.104.05, game version 1.63. Various lengths of time between crashes, and additional issues with smoke rendering very blocky.

The theory that this is effecting 20 series cards seems to hold water.

**Comment by Blisto91** [comment]
@StevenPartridge Hi there. If you suspect the blocky smoke rendering might be a vkd3d-proton could I get you to open a new issue? 
Remember to include a screenshot and graphical settings used if important.

**Comment by HamilcarR** [comment]
The blocky smoke or the display of triangles artifacts like these : 
![cbpk-2077](https://github.com/HansKristian-Work/vkd3d-proton/assets/18567118/54ec724d-c597-470e-b0a0-660a71a3b8a1)
![Screenshot_20230721_015345](https://github.com/HansKristian-Work/vkd3d-proton/assets/18567118/2b879613-187e-4c25-aaf4-09ce160186f4)
![Screenshot_20230721_020626](https://github.com/HansKristian-Work/vkd3d-proton/assets/18567118/24470eb9-b913-4d11-9f17-7a1f0ad0200b)

For me was  caused by proton version or proton ge versions , and sometimes also by drivers version. 
I don't know if it has anything to do with the Xid 31 , but for me this problem came at the same time I think .

**Comment by Blisto91** [comment]
@HamilcarR does this blocky effect still happen on latest Proton Experimental or GE?

**Comment by HamilcarR** [comment]
@Blisto91 I havent played the game in a bit , but last time with the latest proton exp , it was gone , then it came back , I updated the driver , and it was gone again.

**Comment by TarsiSurdi** [comment]
> The blocky smoke or the display of triangles artifacts like these : ![cbpk-2077](https://user-images.githubusercontent.com/18567118/265432820-54ec724d-c597-470e-b0a0-660a71a3b8a1.png) ![Screenshot_20230721_015345](https://user-images.githubusercontent.com/18567118/265432888-2b879613-187e-4c25-aaf4-09ce160186f4.png) ![Screenshot_20230721_020626](https://user-images.githubusercontent.com/18567118/265432903-24470eb9-b913-4d11-9f17-7a1f0ad0200b.png)
> 
> For me was caused by proton version or proton ge versions , and sometimes also by drivers version. I don't know if it has anything to do with the Xid 31 , but for me this problem came at the same time I think .

I've had these graphical issues in the past, but the latest NVIDIA drivers (535.104.05) or Proton Experimental seemed to have fixed it. However, I still experience the intermittent crashes being discussed in this thread.

System Specs: R5 2600X, RTX 2060 6GB, 16 DDR4-2666 - Fedora Workstation 38

**Comment by cpw** [comment]
Saw a link to this issue from the nvidia forums. https://github.com/ValveSoftware/Proton/issues/4450#issuecomment-1738013994 - it seems that this is related to DLSS and any driver more recent than 510.108.03. The problem in cyberpunk seems to be related specifically to the on screen call UI element, in my experience - as soon as the game tries to show one, it locks up on any NVIDIA version where DLSS is possible.

**Comment by extract** [comment]
NVIDIA is looking into this again. Let's hope they find it.

https://forums.developer.nvidia.com/t/cyberpunk-2077-1-62-1-63-crashes/260799/29

From my experience the best way to reproduce this is to fast travel a lot. I have no problems regarding any UI elements causing this exact crash though. Seems anecdotal (but so does the fast travel thing I suppose).

If anyone know a *persistent* method to reproduce it consistently, without any mods etc... then you can add it into the NVIDIA thread linked.

**Comment by haraldkrischner** [comment]
As I also mentioned [here](https://github.com/ValveSoftware/Proton/issues/4450#issuecomment-1773358699), 545.23.06 seems to have finally fixed that Xid 31 error, at least I am playing for 3 hours straight! 🎉
Also [this bug](https://forums.developer.nvidia.com/t/flickering-at-the-top-of-the-screen/256447) I had is gone! 🎉 🎉
